gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/BaseFlowToJobSpecCompiler.java

```diff
@@ -97,35 +97,13 @@ public abstract class BaseFlowToJobSpecCompiler implements SpecCompiler {
 
   private Optional<UserQuotaManager> userQuotaManager;
 
-  public BaseFlowToJobSpecCompiler(Config config){
-    this(config,true);
-  }
-
-  public BaseFlowToJobSpecCompiler(Config config, boolean instrumentationEnabled){
-    this(config, Optional.<Logger>absent(),  true);
-  }
-
-  public BaseFlowToJobSpecCompiler(Config config, Optional<Logger> log){
-    this(config, log,true);
-  }
-
-  public BaseFlowToJobSpecCompiler(Config config, Optional<Logger> log, boolean instrumentationEnabled){
-    this.log = log.isPresent() ? log.get() : LoggerFactory.getLogger(getClass());
-    if (instrumentationEnabled) {
```

Comment:
Is the intention here to remove the use of this variable for checking if instrumentation is enabled? It's still used widely in GaaS modules so there should be justification

Reply:
Arjun did some refactoring before which made instrumentation required I believe. Can you link that change to explain what's happening here?

Reply:
1) yes, in https://github.com/apache/gobblin/pull/3855/files I made some optional fields mandatory. instrumentationEnabled was not one of them, but eventSubmitter was which is related.
2) instrumentationEnabled is usually true in prod, these variables are set false only for doing easier tests.
3) Also, in this particular case, instrumentationEnabled was hard coded to `true` in the above above constructor anyway.